share/spack/templates/container/Dockerfile

```diff
@@ -19,7 +20,7 @@ RUN mkdir {{ paths.environment }} \
 {{ manifest }} > {{ paths.environment }}/spack.yaml
 
 # Install the software, remove unnecessary deps
-RUN cd {{ paths.environment }} && spack env activate . && spack install --fail-fast && spack gc -y
+RUN cd {{ paths.environment }} && spack env activate . && spack concretize && spack env depfile -o Makefile && make -j && spack gc -y
```

Comment:
I wouldn't use `env depfile` by default, cause (a) it will mark everything as explicitly installed (which is a bug we need to fix), and (b) for gmake < 4.4 building `perl` currently swallows all the tokens, so that after perl is installed, things continue single-threaded.

Reply:
Actually (a) was fixed. (b) is still an issue. Btw, use -j$(nproc)